### PR TITLE
Remove modmesh use pyside

### PR DESCRIPTION
**Pull request by j8xixo12**
#210 has mentioned `PySide6` is mandatory, I removed `MODMESH_USE_PYSIDE` in modmesh project.

This patch encounter `PySide6` symbol resolution issue in GitHub Actions `windows` environment.

link to #220

### CODE REVIEW

File: .github/workflows/devbuild.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ jobs:
       run: |
         sudo pip3 install setuptools
         sudo pip3 install numpy pytest flake8
-        sudo pip3 install pyside6 shiboken6
+        sudo pip3 install pyside6==$(qmake6 -query QT_VERSION)
```
Comment (j8xixo12) [inline .github/workflows/devbuild.yml:126]: OS package manager Qt version may not match with default version of pip installed pyside6, therefore need to specific version of pyside6 when using pip install.

ubuntu 22.04 need to use `qmake6` in stead of `qmake`.
ref: https://askubuntu.com/questions/1460242/ubuntu-22-04-with-qt6-qmake-could-not-find-a-qt-installation-of

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -122,6 +122,7 @@ jobs:
 
     - name: make run_viewer_pytest
       run: |
+        export LD_LIBRARY_PATH=$(python3 -c "import sys, os, shiboken6; sys.stdout.write(os.path.dirname(shiboken6.__file__))")
```
Comment (j8xixo12) [inline .github/workflows/lint.yml:125]: Directly specific `shiboken6` path to `LD_LIBRARY_PATH` that allow OS can find `shiboken6` during runtime.

File: cpp/modmesh/toggle/toggle.cpp [hunk 3]
```diff
@@ -64,13 +64,7 @@ Toggle & Toggle::instance()
...
 
 SolidToggle::SolidToggle()
-    : m_use_pyside(
-#ifdef MODMESH_USE_PYSIDE
-          true
-#else
-          false
-#endif
-      )
+    : m_use_pyside(true)
```
Comment (j8xixo12) [inline cpp/modmesh/toggle/toggle.cpp:67]: The reason of I leaves `m_use_pyside` here is modmesh may can select different UI backend in the future, we can use toggle system to select different backend.

File: .github/workflows/devbuild.yml [hunk 4]
```diff
@@ -297,14 +298,18 @@ jobs:
             VERBOSE=1 USE_CLANG_TIDY=OFF \
             BUILD_QT=ON \
             CMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }} \
-            CMAKE_ARGS="-DPYTHON_EXECUTABLE=$(which python3) -DMODMESH_USE_PYSIDE:BOOL=ON"
+            CMAKE_ARGS="-DPYTHON_EXECUTABLE=$(which python3)"
           make buildext VERBOSE=1
 
       - name: make pytest BUILD_QT=ON
         run: |
+          # PySide6 installed by pip will bundle with a prebuilt Qt,
```
Comment (yungyuc) [inline .github/workflows/devbuild.yml:306]: The comments clearly describe why we need to align the Qt library files and are important for future maintenance.  Thanks a lot.

File: cpp/modmesh/toggle/toggle.cpp (hunk 3 above)
Comment (yungyuc) [reply cpp/modmesh/toggle/toggle.cpp:67]: You read my mind.

**Review by yungyuc — APPROVED** [review]
LGTM.  Let me merge.

**Pull request merged**